javacutil/src/org/checkerframework/javacutil/AnnotationUtils.java

```diff
@@ -377,6 +381,92 @@ public static boolean hasInheritedMeta(AnnotationMirror anno) {
+            AnnotationValue aval1 = vals1.get(meth);
+            AnnotationValue aval2 = vals2.get(meth);
+            if (aval1 == null) {
+                aval1 = meth.getDefaultValue();
+            }
+            if (aval2 == null) {
+                aval2 = meth.getDefaultValue();
+            }
+            if (!sameAnnotationValue(aval1, aval2)) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Return true iff the two AnnotationValue objects are the same. Use this instead of
+     * CheckerFrameworkAnnotationValue.equals, which wouldn't get called if the receiver is some
+     * AnnotationValue other than CheckerFrameworkAnnotationValue.
+     */
+    public static boolean sameAnnotationValue(AnnotationValue av1, AnnotationValue av2) {
+        if (av1 == av2) {
+            return true;
+        }
+        if (av1 == null || av2 == null) {
+            return false;
+        }
+        return sameAnnotationValueValue(av1.getValue(), av2.getValue());
+    }
+
+    /**
+     * Return true if the two annotation values are the same. The arguments to this method are
+     * values that are returned by {@code AnnotationValue.getValue()}.
+     */
+    public static boolean sameAnnotationValueValue(Object val1, Object val2) {
+        if (val1 == val2) {
+            return true;
+        }
+
+        // Can't use deepEquals() to compare val1 and val2, because they might have mismatched
+        // AnnotationValue vs. CheckerFrameworkAnnotationValue, and AnnotationValue doesn't override
+        // equals().  So, write my own version of deepEquals().
+        if ((val1 instanceof List<?>) && (val2 instanceof List<?>)) {
+            List<?> list1 = (List<?>) val1;
+            List<?> list2 = (List<?>) val2;
+            if (list1.size() != list2.size()) {
+                return false;
+            }
+            // Don't compare setwise, because order can matter. These mean different things:
+            //   @LTLengthOf(value={"a1","a2"}, offest={"0", "1"})
+            //   @LTLengthOf(value={"a2","a1"}, offest={"0", "1"})
+            for (int i = 0; i < list1.size(); i++) {
+                if (!sameAnnotationValueValue(list1.get(i), list2.get(i))) {
+                    return false;
+                }
+            }
+            return true;
+        } else if ((val1 instanceof AnnotationValue) && (val2 instanceof AnnotationValue)) {
+            return sameAnnotationValue((AnnotationValue) val1, (AnnotationValue) val2);
+        } else if ((val1 instanceof Type.ClassType) && (val2 instanceof Type.ClassType)) {
+            // Type.ClassType does not override equals
+            return sameClassType((Type.ClassType) val1, (Type.ClassType) val2);
+        } else {
+            return Objects.equals(val1, val2);
+        }
+    }
+
+    // Type.ClassType does not override equals
+    public static boolean sameClassType(Type.ClassType t1, Type.ClassType t2) {
```

Comment:
If this is meant as a public method, I would move it to `TypesUtils`.
The existing `org.checkerframework.javacutil.TypesUtils.areSamePrimitiveTypes(TypeMirror, TypeMirror)` has a similar purpose for primitives.
This could be `areSameDeclaredTypes`.
If it is not meant for general consumption, make it private.